src/shared/managers/SubscriptionManager.ts

```diff
@@ -372,20 +372,34 @@ export class SubscriptionManager {
   }
 
   private subscribeSafariPromptPermission(): Promise<string | null> {
-    return new Promise<string>((resolve) => {
-      window.safari.pushNotification.requestPermission(
-        `${SdkEnvironment.getOneSignalApiUrl().toString()}/safari`,
-        this.config.safariWebId,
-        {
-          app_id: this.config.appId,
-        },
-        (response) => {
-          if ((response as any).deviceToken) {
-            resolve((response as any).deviceToken.toLowerCase());
-          } else {
-            resolve(null);
-          }
-        },
+    return new Promise<string | null>((resolve) => {
+      const requestPermission = (url: string) => {
+        window.safari.pushNotification.requestPermission(
+          url,
+          this.config.safariWebId,
+          { app_id: this.config.appId },
+          (response) => {
+            if ((response as any).deviceToken) {
+              resolve((response as any).deviceToken.toLowerCase());
+            } else {
+              // If there's no token and the new URL was used,
+              // retry with the legacy URL
+              if (url.includes('/safari/apps/')) {
+                requestPermission(
```

Comment:
This recursive call makes the code really hard to follow. Could we simply call `await requestPermission(...)`, check if we get a value, then call `await requestPermission(...)` again with the fallback URL?

Reply:
Done.